pkg/tests/schema_pulumi_test.go

```diff
@@ -4159,3 +4160,428 @@ func TestMakeTerraformResultNilVsEmptyMap(t *testing.T) {
+    ~ prov:index/test:Test: (update)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          ~ [0]: {
+                  ~ prop: "value" => output<string>
+                }
+        ]
+Resources:
+    + 1 to create
+    ~ 1 to update
+    2 changes. 1 unchanged
+`))
+		})
+
+		t.Run("unknown object", func(t *testing.T) {
+			program2 := fmt.Sprintf(program, "[\"${auxRes.auxes[0]}\"]")
+			runTest(t, program2, autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    + prov:index/aux:Aux: (create)
+        [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
+    +-prov:index/test:Test: (replace)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          - [0]: {
+                  - prop: "value"
+                }
+          + [0]: output<string>
+        ]
+Resources:
+    + 1 to create
+    +-1 to replace
+    2 changes. 1 unchanged
+`))
+		})
+
+		t.Run("unknown collection", func(t *testing.T) {
+			program2 := fmt.Sprintf(program, "\"${auxRes.auxes}\"")
+			runTest(t, program2, autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    + prov:index/aux:Aux: (create)
+        [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
+    +-prov:index/test:Test: (replace)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      - tests: [
+      -     [0]: {
+              - prop: "value"
+            }
+        ]
+      + tests: output<string>
+Resources:
+    + 1 to create
+    +-1 to replace
+    2 changes. 1 unchanged
+`))
+		})
+	})
+
+	t.Run("set property force new", func(t *testing.T) {
+		resMap := map[string]*schema.Resource{
+			"prov_test": propertyForceNewResource(schema.TypeSet),
+			"prov_aux":  auxResource(schema.TypeSet),
+		}
+
+		tfp := &schema.Provider{ResourcesMap: resMap}
+		bridgedProvider := pulcheck.BridgedProvider(t, "prov", tfp)
+		runTest := func(t *testing.T, program2 string, expectedOutput autogold.Value) {
+			runTest(t, program2, bridgedProvider, expectedOutput)
+		}
+
+		t.Run("unknown plain property", func(t *testing.T) {
+			program2 := fmt.Sprintf(program, "[{prop: \"${auxRes.auxes[0].prop}\"}]")
+			runTest(t, program2, autogold.Expect(`Previewing update (test):
+  pulumi:pulumi:Stack: (same)
+    [urn=urn:pulumi:test::test::pulumi:pulumi:Stack::test-test]
+    + prov:index/aux:Aux: (create)
+        [urn=urn:pulumi:test::test::prov:index/aux:Aux::auxRes]
+    +-prov:index/test:Test: (replace)
+        [id=newid]
+        [urn=urn:pulumi:test::test::prov:index/test:Test::mainRes]
+      ~ tests: [
+          ~ [0]: {
+                  ~ prop: "value" => output<string>
+                }
+        ]
+Resources:
+    + 1 to create
+    +-1 to replace
```

Comment:
Possibly a bug. Slightly edge-casy. When something is unknown we do not know that it will of necessity change, but looks like we assume that.